### ISSUE TITLE
Error while declaring a function named `async`

### ISSUE THREAD

**Issue by rilut** [cherow/cherow#231]
**Code**

```js
cherow.parseScript(`
    let async = function(a){return {bind: "someMethodButIUseString"}};
    async(function (req, res) { }).bind;
`) // errors, but is legal in node

cherow.parseScript(`
    let asyn = function(a){return {bind: "someMethodButIUseString"}};
    asyn(function (req, res) { }).bind;
`) // okay
```

**Expected behavior:**

Should not error while declaring a function named `async`

**Actual behavior:**

SyntaxError: Line 3, column 34: Unexpected token '.'

**Comment by KFlash** [comment]
I can reproduce this one in 1.6.8..

**Comment by Aladdin-ADD** [comment]
is it still happen on master? I'm not sure we have enough manpower to maintain 2 release line.
/cc @KFlash

**Comment by KFlash** [comment]
Confirmed. This will parse with v. 2.0 on Master

**Comment by KFlash** [comment]
Will be fixed in 2.0. Closing
